### PR TITLE
feat/attribute-properties

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: include/geode/basic/attribute_utils.h [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        template < template < typename > class Attribute, typename T >
+        T compute_value( const Attribute< T >& attribute ) const
+        {
+            return AttributeLinearInterpolationImpl< T >::compute(
+                *this, attribute );
+        }
+
+        const absl::FixedArray< index_t > indices_;
+        const absl::FixedArray< double > lambdas_;
+    };
```
Comment (BotellaA) [inline include/geode/basic/attribute_utils.h:124]: or Span? Who should own this information?

File: include/geode/basic/attribute_manager.h [hunk 2]
```diff
@@ -90,17 +90,19 @@ namespace geode
          * attribute, it replaces it by the Attribute corresponding to the
          * attribute.
          * @param[in] name The associated attribute name to look for
+         * @param[in] default_value The default value to use when new attribute
+         * element are created
+         * @param[in] properties The AttributeProperties to use
```
Comment (panquez) [inline include/geode/basic/attribute_manager.h:95]: maybe you can explain what it is

File: include/geode/basic/attribute_manager.h [hunk 3]
```diff
@@ -135,13 +146,17 @@ namespace geode
          * Assign attribute value from other value in the same attribute
          * @param[in] from_element Attribute value to assign
          * @param[in] to_element Where the value is assign
+         * @warning Only affect Attributes created with its AttributeProperties
+         * assignable flag set to true
```
Comment (panquez) [inline include/geode/basic/attribute_manager.h:150]: good :+1:

File: include/geode/basic/attribute_utils.h [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+namespace geode
+{
+    struct AttributeProperties
+    {
+        AttributeProperties() = default;
+
+        AttributeProperties( bool assignable_in, bool interpolable_in )
+            : assignable( assignable_in ), interpolable( interpolable_in )
+        {
+        }
```
Comment (panquez) [inline include/geode/basic/attribute_utils.h:58]:
```suggestion
        AttributeProperties( bool is_assignable, bool is_interpolable )
            : assignable( is_assignable ), interpolable( is_interpolable )
        {
        }
```

File: include/geode/basic/attribute_utils.h [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+#include <geode/basic/passkey.h>
+
+namespace geode
+{
+    struct AttributeLinearInterpolation;
+} // namespace geode
+
+namespace geode
+{
+    struct AttributeProperties
```
Comment (panquez) [inline include/geode/basic/attribute_utils.h:51]: missing doc

**Pull request merged**